src/parentedentity.cpp

```diff
@@ -83,4 +72,9 @@ bool ParentedEntity::hasAncestor(const ParentedEntityPtr &entity) const
     return hasAncestor;
 }
 
+void ParentedEntity::ParentedEntityImpl::setParent(const ParentedEntityPtr &parent)
+{
+    mParent = parent;
+}
```

Comment:
Why are we bothering with this method? I would now expect our code to replace things like:
```c++
rootValueAst->setParent(ast);
component->pFunc()->setParent(newParent);
```
with:
```c++
rootValueAst->mParent = ast;
component->pFunc()->mParent = newParent;
```

Reply:
I think you still want to encapsulate.